crates/apub/src/activities/block/block_user.rs

```diff
@@ -155,19 +155,26 @@ impl ActivityHandler for BlockUser {
     let blocked_person = self.object.dereference(context).await?;
     let target = self.target.dereference(context).await?;
     match target {
-      SiteOrCommunity::Site(_site) => {
-        let blocked_person = Person::update(
-          &mut context.pool(),
-          blocked_person.id,
-          &PersonUpdateForm {
-            banned: Some(true),
-            ban_expires: Some(expires),
-            ..Default::default()
-          },
-        )
-        .await?;
+      SiteOrCommunity::Site(site) => {
+        let site_user_ban_form = SitePersonBanForm {
+          site_id: site.id,
+          person_id: blocked_person.id,
+          expires: Some(expires),
+        };
+        SitePersonBan::ban(&mut context.pool(), &site_user_ban_form).await?;
         if self.remove_data.unwrap_or(false) {
-          remove_user_data(blocked_person.id, context).await?;
+          let user_banned_on_home_instance = verify_domains_match(&site.id(), self.actor.inner())?
+            && verify_domains_match(&site.id(), self.object.inner())?;
+          if user_banned_on_home_instance {
+            remove_user_data(blocked_person.id, context).await?;
+          } else {
+            // Currently, remote site bans federate data removal through corresponding community
+            // bans, because remote site bans were not even stored unless they came from the user's
+            // home instance. We can continue to use community bans to federate data removal for
+            // backwards compatibility, initially but at some point, when most instances have been
+            // upgraded to have the logic of storing remote site bans, we can start doing data
+            // removal here & remove the logic of federating community bans on remote site bans.
```

Comment:
I wouldnt bother with this, the mentioned changes in https://github.com/LemmyNet/lemmy/pull/4464 havent even been released yet and are quite hacky. So theres a chance that it will introduce problems of its own. Better to revert #4464 here and go for the proper solution directly.

Reply:
I agree with this, I'd love to remove my weird logic for banning them from local communities.